modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/transaction/Registry.java

```diff
@@ -36,12 +45,20 @@
 @ThreadSafe
 public class Registry {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Registry.class);
+
     private static final String ASSOCIATED_XID_ERROR_MSG = "Branch still has associated active sessions for xid ";
 
+    private static final String TIMED_OUT_ERROR_MSG = "Transaction timed out for xid ";
+
     private final Map<Xid, Branch> branchMap;
 
+    private final ScheduledExecutorService branchTimeoutExecutorService;
+
     Registry() {
         branchMap = new ConcurrentHashMap<>();
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("DtxBranchTimeoutExecutor-%d").build();
+        this.branchTimeoutExecutorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
```

Comment:
Maybe we should have an object to manage all the threadExecutor in the future.

Reply:
+1. We might need to write our own thread factory as well, if we are getting rid of guava lib